tests/unit-tests/reports/overview/list-table/test-class-sensei-reports-overview-list-table-courses.php

```diff
@@ -7,4 +7,123 @@
  */
 class Sensei_Reports_Overview_List_Table_Courses_Test extends WP_UnitTestCase {
 
+	private static $initial_hook_suffix;
+
+	/**
+	 * Factory for setting up testing data.
+	 *
+	 * @var Sensei_Factory
+	 */
+	private $factory;
+
+	public static function setUpBeforeClass() {
+		parent::setUpBeforeClass();
+		self::$initial_hook_suffix = $GLOBALS['hook_suffix'] ?? null;
+		$GLOBALS['hook_suffix']    = null;
+	}
+
+	public static function tearDownAfterClass() {
+		parent::tearDownAfterClass();
+		$GLOBALS['hook_suffix'] = self::$initial_hook_suffix;
+	}
+
+	/**
+	 * Set up before each test.
+	 */
+	public function setUp() {
+		parent::setUp();
+
+		$this->factory = new Sensei_Factory();
+	}
+
+	/**
+	 * Tear down after each test.
+	 */
+	public function tearDown() {
+		parent::tearDown();
+
+		$this->factory->tearDown();
+	}
+
+	public function testGetColumns_NoCompletionsFound_ReturnsMatchingArray() {
+		$grading = $this->createMock( Sensei_Grading::class );
+		$grading->method( 'get_courses_average_grade' )->willReturn( 2 );
+		$course = $this->createMock( Sensei_Course::class );
+		$course->method( 'get_days_to_completion_total' )->willReturn( 3 );
+		$list_table              = new Sensei_Reports_Overview_List_Table_Courses(
+			$grading,
+			$course,
+			$this->createMock( Sensei_Reports_Overview_Data_Provider_Interface::class )
+		);
+		$list_table->total_items = 1;
+
+		$actual = $list_table->get_columns();
+
+		$expected = [
+			'title'              => 'Course (1)',
+			'last_activity'      => 'Last Activity',
+			'completions'        => 'Completed (0)',
+			'average_percent'    => 'Average Grade (2%)',
+			'days_to_completion' => 'Days to Completion (3)',
+		];
+
+		self::assertSame( $expected, $actual );
+	}
+
+	public function testGetColumns_CompletionsFound_ReturnsMatchingArray() {
+		$user_id   = $this->factory->user->create();
+		$course_id = $this->factory->course->create();
+		Sensei_Utils::update_course_status( $user_id, $course_id, 'complete' );
+		$grading = $this->createMock( Sensei_Grading::class );
+		$grading->method( 'get_courses_average_grade' )->willReturn( 2 );
+		$course = $this->createMock( Sensei_Course::class );
+		$course->method( 'get_days_to_completion_total' )->willReturn( 3 );
+		$list_table              = new Sensei_Reports_Overview_List_Table_Courses(
+			$grading,
+			$course,
+			$this->createMock( Sensei_Reports_Overview_Data_Provider_Interface::class )
+		);
+		$list_table->total_items = 4;
```

Comment:
@merkushin are you using newlines to separate the AAA sections? If so, should we consider using comments instead?

I'm asking because looking at long code that is not using new lines is a bit odd for me. 😄

Reply:
I don't use comments. However, *if you feel it will improve readability, I add them.*

Using newlines is kind of recommended way of separating AAA. But I add newlines sometime when I feel it gives more benefit in terms of clarity. As for this block, I considered it as short enough to not to separate additionally.

Reply:
I'm ok with it in this case, but I'm thinking, maybe long term it would be better to use comments and use newlines to improve readability. What do you think?

Reply:
Yep, I think it is a good idea to have an error-proof solution :)
(Consistently) adding comments sounds like that :)
I'll add them.

Reply:
@m1r0 Added comments and newlines: https://github.com/Automattic/sensei/pull/4938/commits/a3ac9b0f92ccd1960ddcc8761b2ac8751de04a72